File: 2020/25/golang/breaker_test.go
package main 

import (
	"./breaker"
	"testing"
)

func TestShouldPassWithExampleInput(t *testing.T){
	breaker := breaker.ComboBreaker{MOD: 20201227, MAX_LOOP_COUNT: 20201227}
	
	var privateKey, _ = breaker.Decrypt(7, 5764801, 17807724)

	if privateKey == 14897079 {
		t.Log("PASSED breaker.Decrypt(7, 5764801, 17807724)")
	} else {
		t.Errorf("FAILED breaker.Decrypt(7, 5764801, 17807724), expected value %d", 14897079)
	}
}

func TestShouldPassWithGivenInput(t *testing.T){
	breaker := breaker.ComboBreaker{MOD: 20201227, MAX_LOOP_COUNT: 20201227}
	
	door_private_key, _ := breaker.Decrypt(7, 2069194, 16426071)
	card_private_key, _ := breaker.Decrypt(7, 16426071, 2069194)

	if door_private_key == card_private_key {
		t.Log("PASSED breaker.Decrypt(7, 2069194, 16426071)")
	} else {
		t.Errorf("FAILED breaker.Decrypt(7, 2069194, 16426071), expected value %d", 14897079)
	}
}


func TestShouldFailWithBadInput(t *testing.T){
	breaker := breaker.ComboBreaker{MOD: 20201227, MAX_LOOP_COUNT: 20201227}
	
	_, e := breaker.Decrypt(10, 0, 0)

	if e != nil {
		t.Logf("PASSED with captured error %d", e)
	} else {
		t.Errorf("FAILED with no error")
	}
}

File: 2020/25/python/breaker_test.py
import unittest
from combo_breaker import ComboBreaker


class ComboBreakerTests(unittest.TestCase):

    def test_example_from_platform_input(self):
        breaker = ComboBreaker()

        private_key = breaker.decrypt(7, 5764801, 17807724)

        self.assertEqual(private_key, 14897079)

    def test_given_by_platform_input(self):
        breaker = ComboBreaker()

        door_private_key = breaker.decrypt(7, 2069194, 16426071)
        card_private_key = breaker.decrypt(7, 16426071, 2069194)

        self.assertEqual(door_private_key, 11576351)
        self.assertEqual(door_private_key, card_private_key)

    @unittest.expectedFailure
    def test_should_fail_with_incorrect_pubkeys(self):
        breaker = ComboBreaker()

        private_key = breaker.decrypt(10, 0, 0)
        

if __name__ == '__main__':
    unittest.main()